react-demo-app/src/App.tsx

```diff
@@ -0,0 +1,21 @@
+import './App.css';
+import Auth from './components/Auth';
+import Dashboard from './components/Dashboard';
+import { useSelector } from 'react-redux';
+import { IAuthState } from './features/authSlice';
+
+function App() {
+  const isAuthenticated = useSelector((state: IAuthState) => state.auth.isAuthenticated);
```

Comment:
TODO: fix this - this doesn't work how I'd like it to but I haven't really been focused on auth stuff so haven't revisited.